torch/csrc/jit/codegen/cuda/lower_validation.cpp

```diff
@@ -181,6 +258,20 @@ class VectorizeValidator : public OptInDispatch {
         "Issue found in, ",
         tv);
 
+    if (misaligned_vectorize) {
+      if (tv->getMemoryType() == MemoryType::Global) {
+        checkContiguity(validator.domains_, tv);
+      } else if (
+          tv->definition()->getExprType() == ExprType::UnaryOp &&
+          tv->definition()->as<UnaryOp>()->getUnaryOpType() ==
+              UnaryOpType::Set) {
```

Comment:
Shouldn't it just fail if these condition were not met?

Reply:
👍🏼